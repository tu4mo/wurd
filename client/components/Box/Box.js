import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Box = ({ children, className }) => (
  <div className={className}>{children}</div>
)

Box.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
}

const StyledBox = styled(Box)`
  background-color: #fff;
  border-radius: var(--radius-sm);
  box-shadow: var(--shadow);
`

export default StyledBox
