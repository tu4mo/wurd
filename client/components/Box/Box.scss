.box {
  background-color: #fff;
  border-radius: $radius-sm;
  box-shadow: var(--shadow);

  &__section {
    &--has-padding {
      padding: var(--spacing-md);
    }

    &:not(:last-child) {
      border-bottom: 1px solid $color-ultra-light-gray;
    }
  }
}
