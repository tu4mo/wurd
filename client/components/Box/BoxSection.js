import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

const BoxSection = ({ children, className, hasPadding }) => {
  const classNames = classnames(
    'box__section',
    { 'box__section--has-padding': hasPadding },
    className
  )

  return <div className={classNames}>{children}</div>
}

BoxSection.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  hasPadding: PropTypes.bool
}

export default BoxSection
