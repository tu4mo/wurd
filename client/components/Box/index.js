export { default as Box } from './Box'
export { default as BoxSection } from './BoxSection'
