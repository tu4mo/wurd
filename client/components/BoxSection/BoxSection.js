import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'

const BoxSection = ({ children, className, hasPadding }) => (
  <div className={className}>{children}</div>
)

BoxSection.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  hasPadding: PropTypes.bool
}

const StyledBoxSection = styled(BoxSection)`
  ${props =>
    props.hasPadding &&
    css`
      padding: var(--spacing-responsive-md);
    `};

  &:not(:last-child) {
    border-bottom: 1px solid var(--color-ultra-light-gray);
  }
`

export default StyledBoxSection
