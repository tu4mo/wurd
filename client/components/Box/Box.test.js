import React from 'react'
import renderer from 'react-test-renderer'

import Box from './Box'
import BoxSection from './BoxSection'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Box className="test">
        <BoxSection className="test" hasPadding>
          Test
        </BoxSection>
      </Box>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
