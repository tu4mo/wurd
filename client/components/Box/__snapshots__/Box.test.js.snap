// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  background-color: #fff;
  border-radius: var(--radius-sm);
  box-shadow: var(--shadow);
}

.c1 {
  padding: var(--spacing-responsive-md);
}

.c1:not(:last-child) {
  border-bottom: 1px solid var(--color-ultra-light-gray);
}

<div
  className="test c0"
>
  <div
    className="test c1"
  >
    Test
  </div>
</div>
`;
